featurex/diagnostics/__init__.py

```diff
@@ -0,0 +1,123 @@
+    }
+
+
+    ''' Class for holding diagnostics of a design matrix '''
+    def __init__(self, data, columns=None):
+        self.data = data
+
+        cols = self.data.columns if columns == None else columns
+        self.results = {}
+        self.results['Eigenvalues'] = eigenvalues(self.data[cols])
+        self.results['ConditionIndices'] = condition_indices(self.data[cols])
+        self.results['VIFs'] = variance_inflation_factors(self.data[cols])
+        self.results['CorrelationMatrix'] = correlation_matrix(self.data[cols])
+        self.results['RowMahalanobisDistances'] = mahalanobis_distances(self.data[cols])
+        self.results['ColumnMahalanobisDistances'] = mahalanobis_distances(self.data[cols], axis=1)
+        self.results['Variances'] = variances(self.data[cols])
+
+
+    def show(self, stdout=True, plot=False):
+        """
+        Displays diagnostics to the user
+
+        Args:
+            stdout (bool): print results to the console
+            plot (bool): use Seaborn to plot results
+        """
+        if stdout:
+            print('Collinearity summary:')
+            print(pd.concat([self.results['Eigenvalues'],
+                            self.results['ConditionIndices'], 
+                            self.results['VIFs'],
+                            self.results['CorrelationMatrix']],
+                            axis=1))
+
+            print('Outlier summary:')
+            print(self.results['RowMahalanobisDistances'])
+            print(self.results['ColumnMahalanobisDistances'])
+
+            print('Validity summary:')
+            print(self.results['Variances'])
+        
+        if plot:
+            for key, result in self.results.items():
+                if key == 'CorrelationMatrix':
+                    ax = plt.axes()
+                    sns.heatmap(result, cmap='Blues', ax=ax)
+                    ax.set_title(key)
+                    sns.plt.show()
+                else:
+                    result.plot(kind='bar', title=key)
+                    plt.show()
+
+
+    def flag(self, diagnostic, thresh=None):
+        """
+        Returns indices of diagnostic that satisfy (return True from) the 
+        threshold predicate. Will use class-level default threshold if 
+        None provided.
+
+        Args:
+            diagnostic (str): name of the diagnostic
+            thresh (func): threshold function (boolean predicate) to apply to 
+            each element
+        """
+        if thresh is None:
+            thresh = self.defaults[diagnostic]
+
+        result = self.results[diagnostic]
+        if isinstance(result, pd.DataFrame):
+            if diagnostic == 'CorrelationMatrix':
+                result = result.copy()
+                np.fill_diagonal(result.values, 0)
+            return result.applymap(thresh).sum().nonzero()[0]
+        else:
+            return result.apply(thresh).nonzero()[0]
+
+
+    def flag_all(self, thresh_dict={}):
```

Comment:
Nice, I like this idea. One minor tweak you could make that would make it a bit more powerful is to allow the user to indicate which diagnostics to use in the flagging. If you add optional `include` and `exclude` arguments that take lists, then the user can easily select only certain diagnostics to use.